cmd/clusterctl/client/delete_test.go

```diff
@@ -105,33 +108,27 @@ func Test_clusterctlClient_Delete(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			if err := tt.fields.client.Delete(tt.args.options); (err != nil) != tt.wantErr {
-				t.Fatalf("error = %v, wantErr %v", err, tt.wantErr)
-			}
+			err := tt.fields.client.Delete(tt.args.options)
 			if tt.wantErr {
+				g.Expect(err).To(HaveOccurred())
 				return
 			}
 
+			g.Expect(err).NotTo(HaveOccurred())
+
 			proxy := tt.fields.client.clusters["kubeconfig"].Proxy()
 			gotProviders := &clusterctlv1.ProviderList{}
 
 			c, err := proxy.NewClient()
-			if err != nil {
-				t.Fatalf("failed to create client %v", err)
-			}
-
-			if err := c.List(context.TODO(), gotProviders); err != nil {
-				t.Fatalf("failed to read providers %v", err)
-			}
+			g.Expect(err).NotTo(HaveOccurred())
+			g.Expect(c.List(context.TODO(), gotProviders)).To(Succeed())
```

Comment:
I think it's better to use `context.Background()` in tests, as `TODO()` suggests someone is going to fix this later and put something else in place. I know `Background()` is used in the controller tests as an example of us already doing that
```suggestion
			g.Expect(c.List(context.Background(), gotProviders)).To(Succeed())
```

Reply:
done